src/components/manager/ManagerDropBoxContent.js

```diff
@@ -224,30 +275,195 @@ FileDisplay.propTypes = {
+                dispatch(endDropBoxPuttingObjects());
+
+                // Close ourselves (the upload modal)
+                onCancel();
+            })();
+        });
+    };
+
+    return <Modal
+        title="Upload"
+        okButtonProps={{loading: isPutting}}
+        onCancel={onCancel}
+        visible={visible}
+        onOk={onOk}
+    >
+        <FileUploadForm
+            initialUploadFolder={initialUploadFolder}
+            initialUploadFiles={initialUploadFiles}
+            ref={ref => form.current = ref}
+        />
+    </Modal>;
+};
+FileUploadModal.propTypes = {
+    initialUploadFolder: PropTypes.string,
+    initialUploadFiles: PropTypes.arrayOf(PropTypes.instanceOf(File)),
+    onCancel: PropTypes.func,
+    visible: PropTypes.bool,
+};
+
+
+const FileContentsModal = ({selectedFilePath, visible, onCancel}) => {
+    const {tree, isFetching: treeLoading} = useSelector(state => state.dropBox);
+
+    return <Modal
+        visible={visible}
+        title={selectedFilePath ? `${selectedFilePath.split("/").at(-1)} - contents` : ""}
+        width={960}
+        footer={null}
+        onCancel={onCancel}
+    >
+        <FileDisplay file={selectedFilePath} tree={tree} treeLoading={treeLoading} />
+    </Modal>;
+};
+FileContentsModal.propTypes = {
+    selectedFilePath: PropTypes.string,
+    visible: PropTypes.bool,
+    onCancel: PropTypes.func,
+};
+
+
+const InfoDownloadButton = ({disabled, uri}) => (
+    <Button key="download" icon="download" disabled={disabled} onClick={() => {
+        if (uri) {
+            window.open(uri, "_blank");
+        }
+    }}>Download</Button>
+);
+InfoDownloadButton.defaultProps = {
+    disabled: false,
+};
+InfoDownloadButton.propTypes = {
+    disabled: PropTypes.bool,
+    uri: PropTypes.string,
+};
+
+
+const DropBoxInformation = () => (
+    <Alert type="info" showIcon={true} message="About the drop box" description={`
+        The drop box contains files which are not yet ingested into this Bento instance. They are not
+        organized in any particular structure; instead, this serves as a place for incoming data files to be
+        deposited and examined.
+    `} />
+);
+
 
 const ManagerDropBoxContent = () => {
     const history = useHistory();
 
     const dropBoxService = useSelector(state => state.services.dropBoxService);
-    const tree = useSelector(state => state.dropBox.tree);
-    const treeLoading = useSelector(state => state.dropBox.isFetching);
+    const {tree, isFetching: treeLoading} = useSelector(state => state.dropBox);
     const ingestionWorkflows = useSelector(state => workflowsStateToPropsMixin(state).workflows.ingestion);
 
     const filesByPath = useMemo(() => Object.fromEntries(
-        recursivelyFlattenFileTree([], tree).map(f => [f.filePath, f])), [tree]);
+        recursivelyFlattenFileTree([], tree).map(f => [f.relativePath, f])), [tree]);
+
+    const [selectedEntries, setSelectedEntries] = useState(["/"]);
```

Comment:
Can you add a comment that explains the initial state?

Reply:
done!